### PR TITLE
Implements markdown and priview functionality

### LINKED ISSUE #507: Adding Markdown and Preview feature inside "Write Post" within Donut
## Problem to be solved

Currently, there is a button inside the donut platform on the dashboard page which on clicking redirects to write any post. But the actual sense of having the "Write post" comes with proper markdown and preview option. So the purpose of this I to develop the same mechanism and frontend as we see within the GitHub Issue and implement within the **Write Post**.
It should be implemented in the same way as an image depicts:
fbfgrgrg
trhtrhtrhtyh

After it is done, please upload the screenshot for the same.

### PR DESCRIPTION
**Pull request by AuraOfDivinity**
Fixes #507 

Implements Markdown and preview functionality.
fixmarkdownpost

Please review @Rupeshiya @jaskiratsingh2000 @vaibhavdaren @devesh-verma

### CODE REVIEW

File: src/user/dashboard/news-feed/popups/AddPostModal.js [hunk 1]
```diff
@@ -1,66 +1,166 @@
...
+                      ],
+                      textpattern_patterns: [
+                        { start: "#", format: "h1" },
+                        { start: "##", format: "h2" },
+                        { start: "###", format: "h3" },
+                        { start: "####", format: "h4" },
+                        { start: "#####", format: "h5" },
+                        { start: "######", format: "h6" },
+                        { start: "* ", cmd: "InsertUnorderedList" },
+                        { start: "- ", cmd: "InsertUnorderedList" },
```
Comment (vaibhavdaren) [inline src/user/dashboard/news-feed/popups/AddPostModal.js:84]: i am so happy to see this 👍

**Comment by jaskiratsingh2000** [comment]
Glad to see this. Leaving on @vaibhavdaren to check within technicalties

**Comment by Rupeshiya** [comment]
> @AuraOfDivinity But I think this is saving the HTML of the post content in the backend.
> Also during rendering the post content in cards it's showing HTML.
> Please handle this.
> Thanks for this PR Asel. Great wotk 👍

@AuraOfDivinity Is this issue fixed?

**Comment by vaibhavdaren** [comment]
@AuraOfDivinity please fix asap

**Comment by AuraOfDivinity** [comment]
The content is rendered in the exact way it is typed in the editor now. @vaibhavdaren @Rupeshiya 

image

**Comment by Rupeshiya** [comment]
> The content is rendered in the exact way it is typed in the editor now. @vaibhavdaren @Rupeshiya
> 
> image

Cool!!
Looks fine now.

**Pull request merged**